src/index.rs

```diff
@@ -94,6 +95,44 @@ impl IndexWriter {
     fn commit_opstamp(&self) -> u64 {
         self.inner_index_writer.commit_opstamp()
     }
+
+    /// Delete all documents containing a given term.
+    ///
+    /// Args:
+    ///     field_name (str): The field name for which we want to filter deleted docs.
+    ///     field_value (PyAny): Python object with the value we want to filter.
+    ///
+    /// If the field_name is not on the schema raises ValueError exception.
+    /// If the field_value is not supported raises Exception.
+    fn delete_documents(
+        &mut self,
+        field_name: &str,
+        field_value: &PyAny,
+    ) -> PyResult<u64> {
+        if let Some(field) = self.schema.get_field(field_name) {
+            let value = extract_value(field_value)?;
+            let term = match value {
+                Value::Str(text) => Term::from_field_text(field, &text),
+                Value::U64(num) => Term::from_field_u64(field, num),
+                Value::I64(num) => Term::from_field_i64(field, num),
+                Value::F64(num) => Term::from_field_f64(field, num),
+                Value::Date(d) => Term::from_field_date(field, &d),
+                Value::Facet(facet) => Term::from_facet(field, &facet),
```

Comment:
No need to address :

I think this is ok for the moment. This approach will not work if we want to handle auto casting `str` to `Facet` as you suggested in the other PR though.

To do that we need to introduce something like a `fn convert_to_fieldvalue(entry: &FieldEntry, value: &Value) -> FieldValue` or something like that. The logic probably already exists somewhere since we parse Json with it.